access-control-app/src/cmr/access_control/services/acl_util.clj

```diff
@@ -102,3 +103,30 @@
                             (qm/string-condition :target target))
                     identity-type-condition)]
     (get-acls-by-condition context condition)))
+
+(defn sync-entry-titles-concept-ids
+  "If the given ACL is a catalog item acl with a collection identifier that includes concept-ids or
+   entry-titles, return ACL such that both are a unioned with each other for :create action. If action is
+   :update and there are only entry-titles, then sync against entry-titles, otherwise sync against concept-ids.
+   If there are no concept-ids or entry-titles, acl remains unchanged."
+  [context action acl]
+  (if-let [collection-identifier (get-in acl [:catalog-item-identity :collection-identifier])]
+    (let [entry-titles (:entry-titles collection-identifier)
+          concept-ids (:concept-ids collection-identifier)
+          provider-id (get-in acl [:catalog-item-identity :provider-id])
+          colls-from-entry-titles (when (and (seq entry-titles)
```

Comment:
Well we want to keep them in sync, but it would get extremely complicated trying to parse the previous acl and the new acl to see what exactly changed in order to make sure both entry-titles and concept-ids are the same.  The idea is to remove entry-titles completely eventually I believe. So in the case where concept-ids are present in the update I use them as the last say on what should be in both entry-titles and concept-ids.  This simplifies things if our strategy is to try and make work what we are given.  We need entry-titles still because of legacy services.  But I also want to cover the case where the updated acl only has entry-titles, in that case I will use them because if I don't then that results in an empty list for both concept-ids and entry-titles.

We could force users not to use entry-titles at all through validation instead which work fine for MMT  but might break other things.